src/cc_catalog_airflow/dags/provider_api_scripts/brooklyn_museum.py

```diff
@@ -0,0 +1,190 @@
+
+
+def _process_objects_batch(objects_batch):
+    image_count = None
+    for object_ in objects_batch:
+        rights_info = object_.get("rights_type")
+        license_url = _get_license_url(rights_info)
+        logger.debug(license_url)
+        if license_url is not None:
+            id_ = object_.get("id", "")
+            complete_object_data = _get_object_json(
+                endpoint=ENDPOINT+str(id_))
+            if complete_object_data is None:
+                continue
+            image_count = _handle_object_data(data=complete_object_data,
+                                              license_url=license_url)
+
+    return image_count
+
+
+def _handle_object_data(data, license_url):
+    image_count = None
+    image_info = data.get("images", None)
+    if image_info is not None:
+        id_ = data.get("id", "")
+        title = data.get("title", "")
+        foreign_url = f"https://www.brooklynmuseum.org/opencollection/objects/{id_}"
+        metadata = _get_metadata(data)
+        creators = _get_creators(data)
+
+        for image in image_info:
+            foreign_id = image.get("id", "")
+            image_url, thumbnail_url = _get_images(image)
+            if image_url is None:
+                continue
+            height, width = _get_image_sizes(image)
+
+            image_count = image_store.add_item(
+                    foreign_landing_url=foreign_url,
+                    image_url=image_url,
+                    license_url=license_url,
+                    foreign_identifier=foreign_id,
+                    width=width,
+                    height=height,
+                    title=title,
+                    thumbnail_url=thumbnail_url,
+                    meta_data=metadata,
+                    creator=creators
+                )
+
+    return image_count
+
+
+def _get_image_sizes(image):
+    height, width = None, None
+    size_list = image.get("derivatives", "")
+    if type(size_list) is list:
+        size_type = image.get("largest_derivative", "")
+        for size in size_list:
+            if size.get("size", "") == size_type:
+                height = size.get("height", None)
+                width = size.get("width", None)
+    return height, width
+
+
+def _get_license_url(rights_info):
+    license_url = None
+    if "creative commons" in rights_info.get("name").lower():
+        license_url = re.search('https://creativecommons.org/licenses/[^\s]+',
+                                rights_info.get('description'))
+        license_url = license_url.group(0).strip()
+    return license_url
+
+
+def _get_metadata(data):
+    metadata = {}
+    metadata["accession_number"] = data.get("accession_number", "")
+    metadata["date"] = data.get("object_date", "")
+    metadata["description"] = data.get("description", "")
+    metadata["medium"] = data.get("medium", "")
+    metadata["credit_line"] = data.get("credit_line", "")
+    metadata["classification"] = data.get("classification", "")
```

Comment:
It would be better to use the default (Nonetype) fallback on these `get` calls.  Those will eventually be turned into `null` in the DB, and can be more easily cleaned up.